### ISSUE TITLE
Digger may build on air near edge of map

### ISSUE THREAD

**Issue by SimonN** [SimonN/LixD#311]
Lix 0.9.12

See the replay at the end of this post. You can dig near the edge of the map, then build even without air.

**Expected:** Digger falls here, as if there had been air below the dug ground all the time. 

**Probable solution:** Digger should remove ground first (thus altering the beyond-edge-of-map terrain check), then check for air outside of the map.

```
$FILENAME single/lemforum/Lovely/anyway.txt

! 263 0 ASSIGN=BLOCKER 2
! 277 0 ASSIGN=FLOATER 0
! 314 0 ASSIGN=DIGGER 0
! 443 0 ASSIGN=BUILDER 0
```

**Comment by SimonN** [comment]
I haven't yet checked whether the proposed solution in above post is easily applicable. In case it turns out too hard with existing design:

**Fallback solution:**  Bottom of map shouldn't repeat the final row of pixels. Bottom of map should always return empty.

**Comment by SimonN** [comment]
Will be fixed in 0.10.

[See post on Lemmings Forums](https://www.lemmingsforums.net/index.php?topic=3974.msg96794#msg96794)
